Core/clim-core/frames.lisp

```diff
@@ -232,9 +232,20 @@ documentation produced by presentations.")
     (when (frame-manager frame)
       (generate-panes (frame-manager frame) frame)
       (multiple-value-bind (w h) (frame-geometry* frame)
-        (layout-frame frame w h))
+	(layout-frame frame w h))
       (signal 'frame-layout-changed :frame frame))))
 
+(defmethod (setf frame-command-table) :around (new-command-table frame)
+  (flet ((get-menu (x) (slot-value x 'menu)))
+    (let ((prev-menu (get-menu (frame-command-table frame)))
+	  (current-menu (get-menu new-command-table)))
+      (cond
+	((null current-menu) (call-next-method))
+	((null prev-menu) (call-next-method))
+	(t (let ((result (call-next-method)))
+	     (generate-panes *default-frame-manager* *application-frame*)
+	     result))))))
+
 (defmethod generate-panes :before (fm  (frame application-frame))
```

Comment:
Could you use `previous` (full word) for a lexical variable name? Also, this `cond` should be more readable with:
```
(prog1 (call-next-method)
  (when (or (null current-menu)
	    (null previous-menu))
    (generate-panes *default-frame-manager* *application-frame*)))
```